internal/integration/knot/knot_protocol.go

```diff
@@ -179,10 +184,12 @@ func (p *protocol) checkData(device entities.Device) error {
 // Check for device configuration
 func (p *protocol) checkDeviceConfiguration(device entities.Device) error {
 	var ok bool
-	id_pass := 0
+	id_pass := -1
 	// Check if the ids are correct, no repetition
 	for _, data := range device.Config {
-		if data.SensorID != id_pass {
+		if data.Schema.ValueType < 1 || data.Schema.ValueType > 5 {
```

Comment:
What do you think about moving this condition to a function that takes the data.schema.ValueType and returns a boolean? For example, 
func isInvalidValueType(valueType int) bool {
   return valueType < 1 || valueType > 5
}

Also, consider assigning the values 1 and 5 to constants.